RealEstateAgency.AppHost/Program.cs

```diff
@@ -0,0 +1,28 @@
+var builder = DistributedApplication.CreateBuilder(args);
+
+// MongoDB
+var mongodb = builder.AddMongoDB("mongodb")
+    .WithDataVolume("mongodb-data");
+
+var mongoDatabase = mongodb.AddDatabase("realestatedb");
+
+// NATS 
+var nats = builder.AddNats("nats")
+    .WithJetStream()
+    .WithDataVolume("nats-data");
+
+// WebApi
+builder.AddProject<Projects.RealEstateAgency_WebApi>("webapi")
+    .WithReference(mongoDatabase)
+    .WithReference(nats)
+    .WaitFor(mongoDatabase)
+    .WaitFor(nats)
+    .WithExternalHttpEndpoints();
+
+// Generator 
+builder.AddProject<Projects.RealEstateAgency_Generator>("generator")
+    .WithReference(nats)
+    .WaitFor(nats)
```

Comment:
В целом если вам нужны какие то адреса - можно просто через переменные окружения их пробрасывать:
```suggestion
    .WaitFor(nats)
    .WithEnvironment("Nats__Url",nats.GetEndpoint("tcp"))
```

Тут же вам было бы достаточно обратиться к ConnectionString с именем "nats"